### PR TITLE
Org settings 145165791

### PR DESCRIPTION
**Pull request by jjdelc**

### CODE REVIEW

File: source/includes/_endpoint-account.md [hunk 1]
```diff
@@ -47,11 +47,116 @@ GET /account/
...
+    "datasets": "http://app.crunch.io/api/account/datasets/",
+    "applications": "http://app.crunch.io/api/account/applications/"
   }
 }
 ```
 
+##### Applications
+
+```http
+GET /accounts/applications/
```
Comment (nealrichardson) [inline source/includes/_endpoint-account.md]: This should be /account/applications/ right?

File: source/includes/_endpoint-account.md [hunk 2]
```diff
@@ -47,11 +47,114 @@ GET /account/
...
+```
+
+To set/change an application's logo the client needs to make a `multipart/form-data`
+request containing either or both `large` and `small` fields containing the desired
+image files to use. Only account admins are authorized to change this resource.
+
+
+```http
+POST /projects/6c01/icon/ HTTP/1.1
+Content-Disposition: form-data; name="large"; filename="newlogo.jpg"
```
Comment (nealrichardson) [inline source/includes/_endpoint-account.md]: This doesn't seem to match the description: how do I send both large and small in a single request?

File: source/includes/_endpoint-account.md [hunk 3]
```diff
@@ -47,11 +47,114 @@ GET /account/
...
+Content-Type: image/jpeg
+```
+
+```http
+HTTP/1.1 201 Created
+Location: https://app.crunch.io/api/account/applications/1234/
```
Comment (nealrichardson) [inline source/includes/_endpoint-account.md]: Shouldn't the Location be the URL to the logo, not the entity?

File: source/includes/_endpoint-account.md [hunk 4]
```diff
@@ -47,11 +47,114 @@ GET /account/
...
+```
+
+The server will update the images accordingly and validate
+size/dimensions/file format etc.
```
Comment (nealrichardson) [inline source/includes/_endpoint-account.md]: What are the validation criteria/constraints??? What happens if the uploaded file is outside the valid size/dimensions.

File: source/includes/_endpoint-account.md [hunk 5]
```diff
@@ -47,11 +47,114 @@ GET /account/
...
+{
+    "element":"shoji:catalog",
+    "index": {
+        "./mycompany/": {}
+    }
+}
+```
+
+POST a Shoji Entity here to make a new application. The `subdomain` must be
+unique system-wide and must be a valid URL segment; if unavailable or invalid, the server will return a 400 response.
```
Comment (nealrichardson) [inline source/includes/_endpoint-account.md]: What are the validation constraints?

Comment (nealrichardson) [reply source/includes/_endpoint-account.md]: Link to some HTTP spec?

File: source/includes/_endpoint-public.md [hunk 6]
```diff
@@ -1,5 +1,65 @@
...
+            }
+        },
+        "manifest": {}
+    }
+}
+```
+
+Attribute | Type | Description
+----------|------|----------------
+name      | string| Name of the configured application on the given subdomain
```
Comment (nealrichardson) [inline source/includes/_endpoint-public.md]: This kind of table of attributes also belongs in the account applications API section.

File: source/includes/_endpoint-account.md (hunk 2 above)
Comment (jjdelc) [reply source/includes/_endpoint-account.md]: Updating request to include both.

File: source/includes/_endpoint-account.md (hunk 3 above)
Comment (jjdelc) [reply source/includes/_endpoint-account.md]: if you submit both, it should be either of them? 
The benefit of pointing back to the entity is that on it you can see the URLs of all available icons.

File: source/includes/_endpoint-account.md (hunk 4 above)
Comment (jjdelc) [reply source/includes/_endpoint-account.md]: Right now we are only validating file extension, the same as in project icon upload. I did not add an imaging library to the dependencies that would be necessary for this checks to happen. Updating to indicate the "valid file extensions".

File: source/includes/_endpoint-account.md (hunk 5 above)
Comment (jjdelc) [reply source/includes/_endpoint-account.md]: We allow letters, numbers, - according to the spec and a bit of cargo culture (cannot start or end on - and. following python recommendations, no numbers at the beginning). We always force lowercase.

Here's the rfc that indicates numbers letters and -: https://tools.ietf.org/html/rfc952

File: source/includes/_endpoint-public.md (hunk 6 above)
Comment (jjdelc) [reply source/includes/_endpoint-public.md]: Moving

**Pull request merged**